src/compiler/utilitiesPublic.ts

```diff
@@ -1928,8 +1928,8 @@ export function isPropertyAccessOrQualifiedName(node: Node): node is PropertyAcc
 }
 
 /** @internal */
-export function isCallLikeOrFunctionLikeExpression(node: Node): node is CallLikeExpression | SignatureDeclaration {
-    return isCallLikeExpression(node) || isFunctionLike(node);
+export function isCallLikeOrFunctionLikeExpression(node: Node): node is CallLikeExpression | FunctionLikeDeclaration {
+    return isCallLikeExpression(node) || isFunctionLikeDeclaration(node);
 }

```

Comment:
I believe that this reflects the name better and perhaps the change reflects the original intention

Reply:
I guess I don't expect this to return true for function declarations, except those in expression positions... am I missing something?

Reply:
Yee, you are right. I limited this further to only return true for function expressions and arrows

Reply:
Are you trying to limit to functions that can be contextually typed? What about object literal method declarations?

Reply:
> Are you trying to limit to functions that can be contextually typed?

Somewhat. IIRC I meant to limit it to expression-space functions and both of those "sets" are almost the same. I think that either would be OK here (at least for the purpose of my fix).

> What about object literal method declarations?

They are already covered by `isFunctionLikeDeclaration` check